File: languages/java/secure-memory/src/test/java/com/godaddy/asherah/securememory/protectedmemoryimpl/libc/ResourceLimitTest.java
package com.godaddy.asherah.securememory.protectedmemoryimpl.libc;

import static org.junit.jupiter.api.Assertions.*;

import java.util.List;

import org.junit.jupiter.api.Test;

import com.sun.jna.NativeLong;

class ResourceLimitTest {

  @Test
  void testGetFieldOrder() {
    ResourceLimit rLimit = new ResourceLimit();
    List<String> fieldOrder = rLimit.getFieldOrder();
    assertEquals("resourceLimitCurrent", fieldOrder.get(0));
    assertEquals("resourceLimitMaximum", fieldOrder.get(1));
  }

  @Test
  void testZero() {
    ResourceLimit retValue = ResourceLimit.zero();
    NativeLong zeroNativeLong = new NativeLong(0);
    assertEquals(zeroNativeLong, retValue.resourceLimitCurrent);
    assertEquals(zeroNativeLong, retValue.resourceLimitMaximum);
  }

}

File: languages/csharp/SecureMemory/SecureMemory.Tests/ProtectedMemoryImpl/Libc/ResourceLimitTest.cs
using GoDaddy.Asherah.PlatformNative.LP64.Libc;
using Xunit;

namespace GoDaddy.Asherah.SecureMemory.Tests.ProtectedMemoryImpl.Libc
{
    [Collection("Logger Fixture collection")]
    public class ResourceLimitTest
    {
        [Fact]
        private void TestZero()
        {
            rlimit zeroRlimit = rlimit.Zero();
            Assert.Equal(0UL, zeroRlimit.rlim_cur);
            Assert.Equal(0UL, zeroRlimit.rlim_max);
        }
    }
}
